openedx_events/content_authoring/data.py

```diff
@@ -0,0 +1,64 @@
+"""
+Data attributes for events within the architecture subdomain ``content_authoring``.
+
+These attributes follow the form of attr objects specified in OEP-49 data
+pattern.
+
+The attributes for the events come from the CourseDetailView in the LMS, with some unused fields removed
+(see deprecation proposal at https://github.com/openedx/public-engineering/issues/160)
+"""
+from datetime import datetime
+
+import attr
+from opaque_keys.edx.keys import CourseKey
+
+
+@attr.s(frozen=True)
+class CourseScheduleData:
+    """
+    Data describing course scheduling.
+
+    Arguments:
+        start (datetime): course start date
+        end (datetime): course end date
+        pacing (str): 'instructor' or 'self'
```

Comment:
Do you think it's worth having this be an Enum? We have an example of that kinda usage in edx-platform: https://github.com/openedx/edx-platform/blob/master/openedx/core/djangoapps/content/learning_sequences/data.py#L35-L38

Reply:
Not sure about de/serialization of enums, and how it would effect event evolution. In other words, can we simply add enum options under different evolution settings without complaint. 
@ormsbee: Can we punt on this question until later?
@rgraber: Or, do we have a quick way to answer these types of questions?

Reply:
Right now the serializer cannot handle enum fields.

Reply:
We'd probably serialize to string anyhow. It's mostly about making sure people don't create events with typos that get lost. And yeah, I think we can punt this for now.

Reply:
FWIW, in the longer term, I think as long as we strictly add to the enum and don't change old enum values, we can evolve it in a way that's consistent with the evolution of our other data structures.

Reply:
I meant can we make Avro's evolution rules happy, even if it makes sense to us to be able to update. I'm not sure how enums work in Avro, but looks like we are punting for now.

Reply:
Punting.